pkg/controller/jobsync.go

```diff
@@ -229,6 +284,35 @@ func (s *jobSync) setOwnerStatusForInProgressJob(original metav1.Object, job *v1
 	return s.strategy.UpdateOwnerStatus(original, owner)
 }
 
+func (s *jobSync) hasFinalizer(owner metav1.Object) bool {
+	finalizer := s.getFinalizerName()
+	for _, f := range owner.GetFinalizers() {
+		if f == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *jobSync) addFinalizer(original metav1.Object) error {
+	if s.hasFinalizer(original) {
+		return nil
+	}
+	owner := s.strategy.DeepCopyOwner(original)
+	finalizers := sets.NewString(owner.GetFinalizers()...)
+	finalizers.Insert(s.getFinalizerName())
+	owner.SetFinalizers(finalizers.List())
+	return s.strategy.UpdateOwnerStatus(original, owner)
+}
+
+func (s *jobSync) getFinalizerName() string {
+	finalizerName := "openshift/cluster-operator-" + s.jobControl.GetJobPrefix()
+	if !validateFinalizer.MatchString(finalizerName) {
```

Comment:
what is the purpose of this condition? it looks like the finalizerName would never match here.

Reply:
The purpose of this is to ensure that the finalizer name is valid. The finalizer name cannot end in a hyphen, which it would for a job prefix that ends in a hyphen. Given that all of the job prefixes end in hyphens, I suppose it would be better to simply append a valid character to the end in all cases, even when the finalizer may have already been valid to begin with. Is that what you are suggesting?

Reply:
Sorry no, my bad, I was thinking that MatchString would require that the regexp match the entire string, which would require the string to only be a single char :) 
Makes sense what it's doing